framework/src/base/MooseApp.C

```diff
@@ -406,6 +408,21 @@ MooseApp::MooseApp(InputParameters parameters)
     // Sleep to allow time for the debugger to attach
     std::this_thread::sleep_for(std::chrono::seconds(getParam<unsigned int>("stop_for_debugger")));
   }
+
+  // clone the master mesh if provided
+  if (parameters.isParamValid("_master_mesh"))
+  {
+    _action_warehouse.mesh() = parameters.get<const MooseMesh *>("_master_mesh")->safeClone();
+    // looks like safeClone is not cloning all
+    _action_warehouse.mesh()->prepare();
+  }
+
+  if (parameters.isParamValid("_master_displaced_mesh"))
+  {
+    _action_warehouse.displacedMesh() =
+        parameters.get<const MooseMesh *>("_master_displaced_mesh")->safeClone();
```

Comment:
See? It works!

Reply:
It works but partially, not all member variables are cloned and we have to call `prepare` to populate those not cloned. I think we should clone all member variables.

Reply:
I really don't think this implementation is correct. All of this logic is in the constructor of the application, but that's not when we normally build and prepare the mesh. Why not perform the clone during the "setup_mesh" task and then that's it? The "setup_mesh_complete" action would still be responsible for running `prepare` so you wouldn't have that explicit call either. Now the one other piece that would be important would be to short circuit the generators and modifiers, but this would simplify this PR a bit more and make this "less of a special" case. At least the Mesh would still be built and prepared at the normal times.

Reply:
I think the design is that multiapp knows nothing about master, thus it has no way to retrieve the master mesh on its own. We need to pass in mesh pointer anyway. I would not bother mesh is set up before anything but also do not mind the mesh gets cloned in the setup mesh action. In that case, we will have to provide an extra accessor to retrieve the mesh which to me is unnecessary. Ideally, I'd like the parser to not build all these mesh actions at all. I attempted to change the parser but no success due to lack of understanding of it. I do not like the dummy extra interface function in MooseApp in this PR.